compiler/z/codegen/OMRCodeGenerator.cpp

```diff
@@ -3211,6 +3211,101 @@ TR_S390Peephole::AGIReduction()
 
    return performed;
    }
+
+/* For runs with -Xgc:concurrentScavenge on hardware that doesn't support guarded storage facility,
+ * swap LGG/LLGFSG with regular loads and software read barrier sequence that checks if concurrent scavange is in progress
+ * and if the current object pointer is in the evacuate space then calls the GC helper to update the object pointer.
+ */
+
+bool
+TR_S390Peephole::SoftwareReadBarrier()
+   {
+#if defined(OMR_GC_CONCURRENT_SCAVENGER)
+
+   if (TR::Compiler->target.cpu.getS390SupportsGuardedStorageFacility())
+      {
+      return false;
+      }
+
+   auto* concurrentScavangeNotActiveLabel = generateLabelSymbol(_cg);
+   TR::S390RXInstruction *load = (TR::S390RXInstruction*) _cursor;
+   TR::MemoryReference *loadMemRef = generateS390MemoryReference(*load->getMemoryReference(), 0, _cg);
+   TR::Register *loadTargetReg = _cursor->getRegisterOperand(1);
+   TR::Register *vmReg = _cg->getLinkage()->getMethodMetaDataRealRegister();
+   TR::Register *r14Reg = _cg->machine()->getS390RealRegister(_cg->getReturnAddressRegister());
```

Comment:
In the context of OMR this name is misleading. Other consumers of OMR may not use R14 as the return address. Perhaps just calling it `raReg` may be better? Similar changes are needed in comments in this PR below.

Reply:
Changed the variable name to raReg.